sycl/test-e2e/InvokeSimd/Feature/invoke_simd_struct.cpp

```diff
@@ -53,58 +42,76 @@ using namespace sycl::ext::oneapi::experimental;
 namespace esimd = sycl::ext::intel::esimd;
 constexpr int VL = 16;
 
-struct multipliers {
-  int x;
-  int y;
+template <class T> struct solo {
+  T x;
+};
+
+template <class TY, class TX> struct duo {
+  TY y;
+  TX x;
 };
 
-/* Performs the addition of vectors A + B scaled by scalars->x and scalars->y.*/
-__attribute__((always_inline)) esimd::simd<int, VL>
-ESIMD_CALLEE(int *A, esimd::simd<int, VL> b, int i,
-             multipliers *scalars) SYCL_ESIMD_FUNCTION {
-  esimd::simd<int, VL> a;
+template <class TZ, class TY, class TX> struct trio {
+  TZ z;
+  TY y;
+  TX x;
+};
+
+template <class StructTY>
+__attribute__((always_inline)) esimd::simd<float, VL>
+ESIMD_CALLEE(float *A, int i, StructTY S) SYCL_ESIMD_FUNCTION {
+  esimd::simd<float, VL> a;
   a.copy_from(A + i);
-  // Add vectors A + B and scale them by x and y.
-  return a + b * scalars->x * scalars->y;
+  return a * S.x;
 }
 
+template <class StructTY>
 [[intel::device_indirectly_callable]] SYCL_EXTERNAL
-    simd<int, VL> __regcall SIMD_CALLEE(int *A, simd<int, VL> b, int i,
-                                        multipliers *scalars)
-        SYCL_ESIMD_FUNCTION;
-
-int SPMD_CALLEE(int *A, int b, int i, multipliers *scalars) {
-  return A[i] + b * scalars->x * scalars->y;
+    simd<float, VL> __regcall SIMD_CALLEE(float *A, int i,
```

Comment:
Those function names don't tell much to the reader... Can we rename them?

Reply:
There is nothing special about these functions in terms of what happens inside them. Their main purpose is to be called using invoke_simd (hence the name _smth_ + CALLEE) and to take structure as an argument. Most InvokeSimd tests follow this naming scheme.